src/Button/types.ts

```diff
@@ -36,24 +31,23 @@ export type ButtonBaseProps = {
    */
   disabled?: boolean
 } & SxProp &
-  React.ButtonHTMLAttributes<HTMLButtonElement> &
-  StyledButtonProps
+  React.ButtonHTMLAttributes<HTMLButtonElement>
 
 export type ButtonProps = {
   /**
    * The leading icon comes before button content
    */
-  leadingIcon?: React.FunctionComponent<React.PropsWithChildren<IconProps>>
+  leadingIcon?: React.FunctionComponent<React.PropsWithChildren<IconProps>> | null | undefined
```

Comment:
should we add anything 'falsey' as options?   we internally use null and undefined

Reply:
Having `null` here definitely seems useful from the caller side. One quick question for `undefined`, would it already be included without explicitly adding it to the union type?

Reply:
technically yes, but not necessarily.

```
{}
{ leadingIcon: null }
{leadingIcon: () => <></>}
```

would be `{ leadingIcon?: ComponentType | null }`

whereas `{ leadingIcon?: ComponentType | null | undefined }`

includes

```
{ leadingIcon: undefined }
```

due to historical quirks, this case is the same as `?`, but there've been some rumblings about eventually making them explicit in typescript. 

that's enough to say, we could do either.   We could also add false/boolean here to account for things like && returning false?